From 6b55083a5e2e51e23474dace7072093d8fb1f265 Mon Sep 17 00:00:00 2001
From: Ren Jianing <jianing.ren@rock-chips.com>
Date: Fri, 25 Sep 2020 10:02:31 +0800
Subject: [PATCH] usb: gadget: u_audio: fix the risk of strcpy out of bounds

We use strlcpy instead of strcpy to ensure that array won't out
of bounds.

Fixes: a322b6c50d6b (FROMLIST: usb: gadget: f_uac*: Support multiple sampling rates)
Signed-off-by: Ren Jianing <jianing.ren@rock-chips.com>
Change-Id: Iebda94fea94dc929320273c16de996ff0d2d6397
---
 drivers/usb/gadget/function/u_audio.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/usb/gadget/function/u_audio.c b/drivers/usb/gadget/function/u_audio.c
index 74d76748c54e..949629d473b0 100644
--- a/drivers/usb/gadget/function/u_audio.c
+++ b/drivers/usb/gadget/function/u_audio.c
@@ -416,7 +416,7 @@ static struct snd_kcontrol *u_audio_get_ctl(struct g_audio *audio_dev,
 
 	memset(&elem_id, 0, sizeof(elem_id));
 	elem_id.iface = SNDRV_CTL_ELEM_IFACE_PCM;
-	strcpy(elem_id.name, name);
+	strlcpy(elem_id.name, name, sizeof(elem_id.name));
 	return snd_ctl_find_id(audio_dev->uac->card, &elem_id);
 }
 
-- 
2.35.3

